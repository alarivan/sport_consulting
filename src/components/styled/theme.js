import { lighten } from "polished";

const primaryColor = "#1171b7";

const theme = {
  primaryColor,
  primaryColorHover: lighten(0.1, primaryColor),
  borderColor: "#ccc",
  textColor: "#444444"
};

export default theme;
