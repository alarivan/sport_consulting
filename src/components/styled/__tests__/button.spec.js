import React from "react"
import { render } from "@testing-library/react"
import Button from "../Button.css.js"
import theme from "../theme"

describe("Button", () => {
  it("renders correctly", () => {
    const { asFragment, container } = render(<Button>Label</Button>)

    expect(container.firstChild).not.toHaveStyleRule("display", "block")
    expect(container.firstChild).not.toHaveStyleRule("width", "100%")

    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with block modifier", () => {
    const props = { block: true }
    const { asFragment, container } = render(<Button {...props}>Label</Button>)

    expect(container.firstChild).toHaveStyleRule("display", "block")
    expect(container.firstChild).toHaveStyleRule("width", "100%")

    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with theme", () => {
    const props = { theme }
    const { asFragment } = render(<Button {...props}>Label</Button>)

    expect(asFragment()).toMatchSnapshot()
  })
})
