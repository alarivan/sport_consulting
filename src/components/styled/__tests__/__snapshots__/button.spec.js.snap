// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  outline: none;
  border: none;
  border-radius: 3px;
  padding: 0.75rem 2rem;
  cursor: pointer;
  font-weight: 600;
  color: #fff;
  -webkit-transition: background-color 200ms ease-in-out;
  transition: background-color 200ms ease-in-out;
}

.c0:disabled {
  background: #b4b6b8;
  cursor: not-allowed;
}

<button
    class="c0"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`Button renders correctly with block modifier 1`] = `
<DocumentFragment>
  .c0 {
  display: block;
  width: 100%;
  outline: none;
  border: none;
  border-radius: 3px;
  padding: 0.75rem 2rem;
  cursor: pointer;
  font-weight: 600;
  color: #fff;
  -webkit-transition: background-color 200ms ease-in-out;
  transition: background-color 200ms ease-in-out;
}

.c0:disabled {
  background: #b4b6b8;
  cursor: not-allowed;
}

<button
    class="c0"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`Button renders correctly with theme 1`] = `
<DocumentFragment>
  .c0 {
  outline: none;
  border: none;
  border-radius: 3px;
  padding: 0.75rem 2rem;
  cursor: pointer;
  font-weight: 600;
  background-color: #1171b7;
  color: #fff;
  -webkit-transition: background-color 200ms ease-in-out;
  transition: background-color 200ms ease-in-out;
}

.c0:not(:disabled):hover,
.c0:not(:disabled):focus,
.c0:not(:disabled):active {
  background-color: #158ee6;
}

.c0:disabled {
  background: #b4b6b8;
  cursor: not-allowed;
}

<button
    class="c0"
  >
    Label
  </button>
</DocumentFragment>
`;
